test/rest-adapter.test.js

```diff
@@ -17,6 +17,10 @@ var factory = require('./helpers/shared-objects-factory.js');
 describe('RestAdapter', function() {
   var remotes;
 
+  before(function() {
+    process.env.NO_DEPRECATION = 'strong-remoting';
+  });
+
```

Comment:
>I am proposing the following:

>* Rework all tests using deprecated methods to use the new official methods
* In unit-tests that are testing specifically these deprecated methods, temporarily disable deprecation messages via https://github.com/dougwilson/nodejs-depd#processenvno_deprecation

I reworked on [where](https://github.com/strongloop/strong-remoting/pull/305/files#diff-bc246aed6f72bde8d7cc584ef435426aR166) we were using deprecated `getHttpMethod` and `getPath` to use our new official methods, but [here](https://github.com/strongloop/strong-remoting/blob/ac3093dcfbb787977ca0229b0f672703859e52e1/test/rest-adapter.test.js#L220-L245) that we we are explicitly testing these deprecated methods, I should just have them untouched, and need to disable deprecation messages temporarily. However, I tried `process.env.NO_DEPRECATION = 'strong-remoting';` but still it shows the deprecation messages...

Also it is worth mentioning that there are many other deprecated messages for other methods (not `getHttpMethod` or `getPath`)which are shown when you run tests for `strong-remoting`, I did not touch them because I thought it is out of scope of this PR.